premerge/cluster-management.md

```diff
@@ -98,3 +98,107 @@ Setting the cluster up for the first time is more involved as there are certain
 resources where terraform is unable to handle explicit dependencies. This means
 that we have to set up the GKE cluster before we setup any of the Kubernetes
 resources as otherwise the Terraform Kubernetes provider will error out.
+
+## Upgrading/Resetting Github ARC
+
+Updating and resetting the Github Actions Runner Controller (ARC) within the
+cluster involves largely the same process. Some special considerations need
+to be made with how ARC interacts with kubernetes. The process involves
+uninstalling the runner scale set charts, deleting the namespaces to ensure
+everything is properly cleaned up, optionally bumping the version number if
+this is a version upgrade, and then reinstalling the charts to get the cluster
+back to accepting production jobs.
+
+It is important to not just blindly delete controller pods or namespaces as
+this (at least empirically) can interrupt the state and custom resources that
+ARC manages, then requiring a costly full uninstallation and reinstallation of
+at least a runner scale set.
+
```

Comment:
Can we get a bit here about impact on any running build jobs?

Reply:
Best practice is to just uninstall the helm charts. The jobs currently running will still finish up, and ARC will not accept any more jobs. If you set `maxRunners` to zero, ARC might accept some jobs but never start executing them. I don't think that would cause any problems in our configuration, but for HA configurations with multiple clusters, it can cause non-seamless failover from what I understand.

https://www.kenmuse.com/blog/strategies-for-upgrading-arc/

Reply:
Added some comments on this.